src/main/java/io/vertx/openapi/contract/OpenAPIVersion.java

```diff
@@ -75,16 +76,42 @@ public static OpenAPIVersion fromContract(JsonObject contract) {
     }
   }
 
-  public Future<OutputUnit> validate(Vertx vertx, SchemaRepository repo, JsonObject contract) {
-    return vertx.executeBlocking(p -> p.complete(repo.validator(mainSchemaFile).validate(contract)));
+  public Future<OutputUnit> validateContract(Vertx vertx, SchemaRepository repo, JsonObject contract) {
+    return vertx.executeBlocking(() -> repo.validator(mainSchemaFile).validate(contract));
+  }
+
+  /**
+   * Validates additional contract files against the openapi schema. If validations fails, try to validate against the
+   * json schema specifications only.
+   *
+   * @param vertx                   The related Vert.x instance.
+   * @param repo                    The SchemaRepository to do the validations with.
+   * @param file                    The additional json contract to validate.
+   */
+  public Future<Void> validateAdditionalContractFiles(Vertx vertx, SchemaRepository repo, JsonObject file) {
+    return validateContract(vertx, repo, file)
+      .compose(this::checkOutputUnit)
+      .recover((t) -> {
+        //Schema validation failed using the openapi spec, so instead try to use json schema spec only.
+        return checkOutputUnit(repo.validator(draft.getIdentifier()).validate(file));
```

Comment:
We should run this valiation in executeBlocking, because it could block the EL

Reply:
Good point, i've updated this to be in an executeBlocking().